### ISSUE TITLE
Comparing an enum element to a value results in compiler error

### ISSUE THREAD

**Issue by ben-albrecht** [chapel-lang/chapel#6208]
### Summary of Problem

Comparing an enum element to a value results in the following `==` overload resolving:

```chapel
   inline proc ==(param a: int(?w), param b: int(w)) param return __primitive("==", a, b);
```

Which triggers the following error in `postFold.cpp`:

```c
   USR_FATAL_CONT(call, "'param' functions cannot return non-'param' values");
```

This particular error propagates to several other operations on enums,
including `%` (modulus), which is how this error was surfaced (in #6187).

### Steps to Reproduce

**Associated Future Test(s):**

[`test/types/enum/compare.chpl`](https://github.com/chapel-lang/chapel/tree/master/test/types/enum/compare.chpl) #6207

**Comment by s-geronimoanderson** [comment]
This issue affects the DynamicIters module because it has a conditional concerning the adaptive iterator stealing method:

```chapel
if (methodStealing==Method.Whole
    || methodStealing==Method.WholeTail
    || (methodStealing==Method.RoundRobin && stealFailed))
then { ... }
```

In the meantime, one must set `methodStealing` for the adaptive iterator with the appropriate enum element (not with an integer). For example:

```shell
chpl --set methodStealing=Method.RoundRobin ...
```

**Comment by daviditen** [comment]
The problem here seems to be that when an `enum` is passed to a formal argument expecting a `param int` it is instantiated as an `enum` instead of the equivalent `int` value.

The following simplified example prints "one", but it should print "1".

```
enum Number {
  one = 1
}

proc paramArg(param a: int) {
  writeln(a);
}

paramArg(Number.one);
```

**Comment by bradcray** [comment]
It appears that @mppf fixed this issue in PR #7659 but failed to close this issue or remove the .bad file.  Leaving it open for now in hopes that someone else will confirm (and remove the .bad file).

**Comment by mppf** [comment]
Right, I un-futured this in PR #7659. I'll close this issue and I've opened PR #8564 to remove the .bad file.
